### ISSUE TITLE
scatterplot  - logarithmic y axis changes y_lim unexpectedly

### ISSUE THREAD

**Issue by theitush** [mwaskom/seaborn#2365]
So, this: 
```python
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
 
fig, ax = plt.subplots(figsize=(10, 10), ncols=1, nrows=1) 
df = pd.DataFrame(data={'x': [1, 2, 3], 'y': [1, 0.1, 0.01]})
sns.scatterplot(x="x", y="y", data=df, ax=ax)
ax.set_yscale('log')
```
Works great:
![image](https://user-images.githubusercontent.com/46968791/100544653-ed553b80-325f-11eb-8b42-32f7534b8499.png)

but y_lim get's weird when using slightly different data:
```python
df = pd.DataFrame(data={'x': [1, 2, 3], 'y': [1, 0.1, 0.001]})
```
![image](https://user-images.githubusercontent.com/46968791/100544665-fba35780-325f-11eb-9a95-fc46e390b46e.png)

This is easily fixed by setting the y_lim manually but still a confusing and surprising issue from an otherwise almost perfect package!

Also, it seems to depend on the figure size.. so bigger figures can tolerate more extreme y values.

Thanks!

**Comment by mwaskom** [comment]
The axis scaling is handled by matplotlib, a matplotlib-only version of your example would be

```python
fig, ax = plt.subplots(figsize=(10, 10), ncols=1, nrows=1) 
df = pd.DataFrame(data={'x': [1, 2, 3], 'y': [1, 0.1, 0.001]})
ax.scatter(x="x", y="y", data=df)
ax.set_yscale('log')
```

I cannot, however, reproduce your issue with matplotlib 3.3.1, either using seaborn or not.

You didn't mention your matplotlib version, but I suspect you've encountered https://github.com/matplotlib/matplotlib/issues/6015, which was fixed in matplotlib 3.2.

Either way, this is not a seaborn problem, so I will close the issue here.

**Comment by theitush** [comment]
huh, looks like you are correct!

Sorry and thank you.
